include/mtxclient/crypto/client.hpp

```diff
@@ -188,6 +194,14 @@ class OlmClient : public std::enable_shared_from_this<OlmClient>
         OlmAccount *account() { return account_.get(); }
         OlmUtility *utility() { return utility_.get(); }
 
+        //! SAS related stuff
+        //! this creates a unique pointer of class OlmSAS
+        SASPtr sas_init();
+        std::string sas_get_pub_key(OlmSAS *sas);
+        void set_their_key(OlmSAS *sas, std::string their_public_key);
+        std::vector<int> generate_bytes_decimal(OlmSAS *sas, std::string info);
+        std::vector<int> generate_bytes_emoji(OlmSAS *sas, std::string info);
```

Comment:
I think it may be cleaner to wrap this in a custom class. This could maybe look like this:

```
struct SAS
{
        SAS(std::string their_public_key); // constructor, does init and set_their_key
        std::string public_key(); // returns public key
        std::vector<int> generate_bytes_decimal(OlmSAS *sas, std::string info);
        std::vector<int> generate_bytes_emoji(OlmSAS *sas, std::string info);
private:
        SASPtr sas;
};

Reply:
Okay

Reply:
But ig in the constructor I wont be taking ```their_public_key``` as parameter cause when I use it ```nheko``` I would get their public key in the later stages.

Reply:
Well, I'm not sure, how you intend to use it, but I thought you would only allocate the SAS object, when you actually want to use it and you may have the public key available in that case already. If that doesn't match with what you intend to do, then yes, taking the public key separately makes sense :3

Reply:
I was planning to create a SAS object for every flow and then when I get the ```key.verification.start``` that contains the public key of the other user I would assign it then.

Reply:
Yeah, makes sense. You also need to set the key alter for your tests.